### DISCUSSION TITLE
Blog: Incrementally Adopting Next.js

### DISCUSSION THREAD

**Discussion by timneutkens** [vercel/next.js#19735] [Show and tell]
Next.js has been designed for gradual adoption. With Next.js, you can continue using your existing code and add as much (or as little) React as you need. By starting small and incrementally adding more pages, you can prevent derailing feature work by avoiding a complete rewrite.

Read more: https://nextjs.org/blog/incremental-adoption

**Comment by dasblitz** [comment]
I have a question about this in combination with `fallback: 'blocking'` which might be a bug or me misunderstanding:
https://github.com/vercel/next.js/discussions/21348

**Reply by 6264737658** [reply]
No
